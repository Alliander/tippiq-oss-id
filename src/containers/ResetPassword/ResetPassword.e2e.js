import secureRandomString from 'secure-random-string';

import { page as registerPage } from '../Register/Register.e2e';

export const page = {
  isLoaded: () => protractor.ExpectedConditions.presenceOf(element(by.id('page-reset-password'))),
  heading: () => element(by.css('h1')).getText(),
  submit: () => element(by.id('passwordSubmitButton')).click(),
  passwordField: () => element(by.id('password')),
  passwordError: () => element(by.id('password'))
    .element(by.xpath('following-sibling::div')).getText(),
  result: () => element(by.id('successMessage')),
  residentTitle: () => element(by.css('.resident-title')).getText(),
};

describe('Reset password container', () => {
  const randomString = secureRandomString();
  it('should set a new password', () => {
    // Register user to create token
    registerPage.open();

    registerPage.emailField().sendKeys(`${randomString}@example.com`);
    registerPage.passwordField().sendKeys('test1234ABC');
    registerPage.submit();

    browser.wait(protractor.ExpectedConditions.presenceOf(registerPage.result()), 10000);

    // Reset password
    browser.get('/nieuw-wachtwoord-instellen');
    browser.wait(page.isLoaded(), 10000);
    expect(page.heading()).toEqual('Nieuw wachtwoord instellen');

    // Check required field
    page.submit();
    expect(page.passwordError()).toEqual('Verplicht');

    // Check submit
    page.passwordField().sendKeys('test1234ABC');
    page.submit();

    browser.wait(protractor.ExpectedConditions.presenceOf(page.result()), 10000);
    expect(page.result().getText()).toEqual('Gelukt! Je nieuwe wachtwoord is opgeslagen.');

    // Check user is logged in
    expect(page.residentTitle().getText()).toEqual(`${randomString.substring(0, 17)}...`);
  });
});
