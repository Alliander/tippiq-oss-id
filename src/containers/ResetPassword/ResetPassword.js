/**
 * Reset password container.
 * @module components/ResetPassword
 */

import React, { Component, PropTypes } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';
import { Grid, Row, Col, Button } from 'react-bootstrap';
import { get } from 'lodash';

import { ChangePasswordForm, PageTitle } from '../../components';
import { changePassword, tokenLogin } from '../../actions';

/**
 * Reset password container class.
 * @class ResetPassword
 * @extends Component
 */
export class ResetPassword extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    changePassword: PropTypes.func,
    location: PropTypes.object,
    password: PropTypes.object.isRequired,
    tokenLogin: PropTypes.func.isRequired,
  }

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentWillMount() {
    const token = get(this.props, 'location.query.token');
    if (token) {
      this.setState({ token });
    }
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    const { password } = nextProps;
    if (password.token) {
      this.props.tokenLogin(password.token);
    }
  }

  /**
   * Handler for submit action.
   * @method handleSubmit
   * @param {Object} data Form data.
   * @returns {undefined}
   */
  handleSubmit = (data) => {
    this.props.changePassword(data.password, get(this.state, 'token'));
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { password } = this.props;
    const errorMessage = password.error ? password.error.message : null;
    const successView = password.save && password.save.success ? (
      <Grid fluid>
        <Row>
          <Col xs={9}>
            <PageTitle>
              <span id="successMessage">Gelukt! Je nieuwe wachtwoord is opgeslagen.</span>
            </PageTitle>
          </Col>
        </Row>
        <Row>
          <Col xs={12}>
            <br />
            <p>
              Vanaf nu kun je inloggen met je nieuwe wachtwoord.
            </p>
            <Button
              onClick={() => browserHistory.push('/')}
              bsStyle="primary"
            >Ga naar mijn gegevens</Button>
          </Col>
        </Row>
      </Grid>) : null;

    return (
      <div id="page-reset-password">
        <Helmet title="Mijn Account | Wachtwoord wijzigen" />
        { successView ||
          <Grid fluid>
            <Row>
              <Col xs={12}>
                <PageTitle>Nieuw wachtwoord instellen</PageTitle>
              </Col>
            </Row>
            <Row>
              <Col xs={12}>
                <p>
                  Kies een nieuw wachtwoord voor je Tippiq <span className="text-primary">Account</span>.
                </p>
                <br />
              </Col>
            </Row>
            <Row>
              <Col xs={12}>
                <ChangePasswordForm
                  onSubmit={this.handleSubmit}
                  errorMessage={errorMessage}
                />
              </Col>
            </Row>
          </Grid>
        }
      </div>
    );
  }
}

export default connect(
  state => ({
    password: state.password,
  }), dispatch => bindActionCreators({ changePassword, tokenLogin }, dispatch)
)(ResetPassword);
