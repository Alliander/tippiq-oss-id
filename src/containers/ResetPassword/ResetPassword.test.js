import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ResetPassword } from './ResetPassword';

test('Basic render snapshot test', () => {
  const component = shallow(<ResetPassword
    error={'error'}
    changePassword={() => {}}
    location={{ query: '' }}
    password={{
      save: {
        success: '',
      },
      error: {
        message: '',
      },
    }}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

