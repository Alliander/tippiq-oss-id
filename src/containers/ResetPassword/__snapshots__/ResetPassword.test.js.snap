// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<div
  id="page-reset-password"
>
  <HelmetWrapper
    title="Mijn Account | Wachtwoord wijzigen"
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <PageTitle>
          Nieuw wachtwoord instellen
        </PageTitle>
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <p>
          Kies een nieuw wachtwoord voor je Tippiq 
          <span
            className="text-primary"
          >
            Account
          </span>
          .
        </p>
        <br />
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <ConnectedForm
          errorMessage=""
          onSubmit={[Function]}
        />
      </Col>
    </Row>
  </Grid>
</div>
`;
